docs/README.md

```diff
@@ -0,0 +1,157 @@
+  - TLS connection is also required.
+- Delta management is not required for server side.
+- To restrict access to the server, cookie can be used.
+- All files to be updated are verified by the hash included in the metadata, and the metadata is also verified by X.509 certificate locally installed.
+- Transfer data is encrypted by TLS
+- Multiple ECU support
+- By the internal proxy cache mechanism, the cache can be used for the download requests to the same file from multiple ECU.
+
+## License
+
+OTA client is licensed under the Apache License, Version 2.0.
+
+## OTA client setup
+
+### Requirements
+
+- supported boot loader
+  - GRUB
+  - [CBoot](https://docs.nvidia.com/jetson/archives/l4t-archived/l4t-3271/index.html#page/Tegra%20Linux%20Driver%20Package%20Development%20Guide/bootflow_jetson_xavier.html#wwpID0E0JB0HA)
+
+- runtime
+  - python3.8 (or higher)
+
+### Partitioning
+
+TODO
+
+### Configurations
+
+OTA client can update a single ECU or multiple ECUs and is installed for each ECU.
+There two types of ECU, Main ECU - receives user request, Secondary ECUs - receives request from Main ECU. One or multiple Secondary ECUs can also have Secondary ECUs.
+Note that if only one ECU is updated, there is not Secondary ECUs.
+
+#### ecu\_info.yml
+
+ecu_info.yml is the setting file for ECU configuration.
+
+##### File path
+
+/boot/ota/ecu_info.yml
+
+##### Entries
+
+- format_version (string, required)
+
+  This field specifies the ecu_info.yml format.  
+  Currently this field is not used but `1` should be specified for future use.
+
+- ecu_id (string, required)
+
+  This field specifies ECU id and that should be unique in all EUCs.
+
+- ip_addr (string, optional)
+
+  This field specifies this ota client's IP address.  
+  If this field is not specified, "localhost" is used.  
+  NOTE: this IP address is used for the local server bind address.
+
+- secondaries (array, optional)
+
+  This field specifies list of secondary ECUs information.  
+  If this field is not specified, it is treated as if secondary ecu doesn't exist.
+
+  - ecu_id (string, required)
+    ecu id of secondary ECU
+
+  - ip_addr (string, required)
+    IP address of secondary ECU.
+
+- available_ecu_ids (optional)
+  This field specifies a list of ECU id which indicates the target ECU of the ota update.  
+  Only the main ECU should have this information.  
+  If this field doesn't exist, ecu_id value is used as this value.  
+
+  NOTE: The difference between secondaries and available_ecu_ids:  
+  `secondaries` field is the information for listing children's ECUs, not for
+  listing grandchildren. `available_ecu_ids` filed is the information for listing
+  all ECUs. If the child ecu has its own children, and if that child doesn't
+  respond to the parent's request, the grandchildren's information can't be
+  retrieved by the request. That is the reason why available_ecu_ids field is
+  needed.
```

Comment:
what about emphasizing that `secondaries` means the `directly connected children ECUs`, and the `available_ecu_ids` consists of all the children ECUs and itself. 

For example:
`secondaries` lists the directly connected children ECUs, `available_ecu_ids` consists of all children ECUs(including directly connected and indirect connected).